libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -66,7 +66,9 @@ uint32_t ByteBuffer::space(void) const
         ret = size;
     }
 
-    ret += _head - tail - 1;
+	if(_head != tail) {
```

Comment:
This should always be true because if _head equals to tail, than the previous if would have returned already, wouldn't?